src/harness/reference_models/interference/interference.py

```diff
@@ -413,57 +394,73 @@ def computeInterferenceFssCochannel(cbsd_grant, constraint, fss_info, max_eirp):
+  # range is considered and calculation is done on both segments
+  elif ((offset - 100*MHZ) <= cbsd_grant.low_frequency and
+        constraint.high_frequency >= cbsd_grant.high_frequency):
+    # Get the total of 1MHZ frequencies overlapping within the segment1 range
+    num_one_mhz_bin_seg1 = int((cbsd_grant.high_frequency - offset)/MHZ)
+    fss_mask_attenuation = 0
+    for i in range (0, num_one_mhz_bin_seg1):
+      fss_mask_attenuation +=10**(-(((constraint.high_frequency - offset)/int(MHZ)) - 0.5 -i)*0.6 - 0.5)/10
+    fss_mask_attenuation_seg1 = fss_mask_attenuation/num_one_mhz_bin_seg1
+    fss_mask_attenuation = 0
+    # Get the total of 1MHZ frequencies overlapping within the segment2 range
+    num_one_mhz_bin_seg2 = int((offset - cbsd_grant.low_frequency)/MHZ)
+    for i in range (0, num_one_mhz_bin_seg2):
+      fss_mask_attenuation +=10**(-(((offset - cbsd_grant.low_frequency)/int(MHZ)) - 0.5 -i)*0.25 - 30.5)/10
+    fss_mask_attenuation_seg2 = fss_mask_attenuation/num_one_mhz_bin_seg2
 
-  fss_mask_loss = 0
-
-  # if lower edge of the FSS passband is less than CBSD grant
-  # lowFrequency and highFrequency
-  if (constraint.low_frequency < cbsd_grant.low_frequency and
-         constraint.low_frequency < cbsd_grant.high_frequency):
-    fss_mask_loss = 0.5
-
-  # if CBSD grant lowFrequency and highFrequency is less than
-  # 50MHz offset from the FSS passband lower edge
-  elif (cbsd_grant.low_frequency < offset and
-     cbsd_grant.high_frequency < offset):
-    fss_mask_loss = linearToDb((cbsd_freq_range / MHZ) * 0.25)
-
-  # if CBSD grant lowFrequency is less than 50MHz offset and
-  # highFrequency is greater than 50MHz offset
-  elif (cbsd_grant.low_frequency < offset and
-            cbsd_grant.high_frequency > offset):
-    low_freq_mask_loss = linearToDb(((offset - cbsd_grant.low_frequency) / MHZ) * 0.25)
-    fss_mask_loss = low_freq_mask_loss + linearToDb(((
-        cbsd_grant.high_frequency - offset) / MHZ) * 0.6)
-
-  # if FSS Passband lower edge frequency is grater than CBSD grant
-  # lowFrequency and highFrequency and
-  # CBSD grand low and high frequencies are greater than 50MHz offset
-  elif (constraint.low_frequency > cbsd_grant.low_frequency and
-      constraint.low_frequency > cbsd_grant.high_frequency and
-      cbsd_grant.low_frequency > offset and
-             cbsd_grant.high_frequency > offset):
-    fss_mask_loss = linearToDb((cbsd_freq_range / MHZ) * 0.6)
-
-  return fss_mask_loss
+  else:
+    raise Exception("Fss Mask is not calculated for grants falling within band 3700 to 4200")
 
+  # Calculate the final attenuation value by adding the overlap mask values calulated
+  # in both the segments
+  fss_mask_attenuation = fss_mask_attenuation_seg1+fss_mask_attenuation_seg2
+  return fss_mask_attenuation
 
+   
 def computeInterferenceFssBlocking(cbsd_grant, constraint, fss_info, max_eirp):
   """Computes interference that a grant causes to a FSS protection point.
-
```

Comment:
Again why removing the blank lines here and everywhere else. This goes against the convention 
used in the test harness code.

Reply:
Added   back the removed new lines